src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/LimitedPartnershipService.java

```diff
@@ -22,12 +25,19 @@ public LimitedPartnershipService(LimitedPartnershipMapper mapper, LimitedPartner
         this.repository = repository;
     }
 
-    public String createLimitedPartnership(LimitedPartnershipSubmissionDto limitedPartnershipSubmissionDto, String requestId, String userId) {
+    public String createLimitedPartnership(Transaction transaction, LimitedPartnershipSubmissionDto limitedPartnershipSubmissionDto, String requestId, String userId) {
         ApiLogger.debug("Called createLimitedPartnership(...)");
 
         LimitedPartnershipSubmissionDao dao = mapper.dtoToDao(limitedPartnershipSubmissionDto);
         dao.setCreatedAt(LocalDateTime.now());
         dao.setUserId(userId);
+
+        // Create the self-link
+        String selfLink = String.format("/transactions/%s/limited-partnership-submissions/%s", transaction.getId(), dao.getId());
```

Comment:
This doesn't match the format of the self link specified in the sub-task.

Reply:
edited to use correct format